osf/management/commands/migrate_education_employment.py

```diff
@@ -0,0 +1,182 @@
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import connection
+
+from framework.celery_tasks import app as celery_app
+from osf.models import OSFUser, UserEducation, UserEmployment
+from osf.models.base import generate_object_id
+
+logger = logging.getLogger(__name__)
+
+
+def populate_new_models(rows, dry_run):
+
+    users_with_education = OSFUser.objects.raw('''SELECT id, schools FROM osf_osfuser where schools <> '{}' LIMIT %s''', [rows])
+    set_model(users_with_education, 'schools', dry_run)
+
+    users_with_employment = OSFUser.objects.raw('''SELECT id, jobs FROM osf_osfuser where jobs <> '{}' LIMIT %s''', [rows])
+    set_model(users_with_employment, 'jobs', dry_run)
+
+
+def parse_model_datetime(month, year):
+    return datetime.datetime.strptime('{} {}'.format(month, year), '%m %Y')
+
+
+def get_dates(entry):
+    start_year = entry.get('startYear', None)
+    start_month = entry['startMonth'] if start_year else None
+    if start_year and start_month:
+        start_date = parse_model_datetime(start_month, start_year)
+    else:
+        start_date = None
+
+    end_year = entry.get('endYear', None)
+    end_month = entry['endMonth'] if end_year else None
+    if end_year and end_month:
+        end_date = parse_model_datetime(end_month, end_year)
+    else:
+        end_date = None
+
+    return start_date, end_date
+
+
+def set_model(queryset, attribute_name, dry_run):
+    if attribute_name == 'schools':
+        query = 'INSERT INTO osf_usereducation (_id, user_id, ongoing, degree, department, institution, end_date, start_date, created, modified, _order) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)'
+
+    if attribute_name == 'jobs':
+        query = 'INSERT INTO osf_useremployment (_id, user_id, ongoing, title, department, institution, end_date, start_date, created, modified, _order) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)'
+
+    with connection.cursor() as cursor:
+        for user in queryset:
+            entry = getattr(user, attribute_name)
+            for order, data in enumerate(entry):
```

Comment:
Why did you decide to go with this rather than the sql that would get the whole table at a time?

Reply:
Not sure what you mean, here I have to iterate over the list of dicts because one list is multiple rows in the table.

Reply:
there is a second draft pr attached to the ticket that I did which had a comment with some sql that splits apart the array ifnjson into rows.

Reply:
there is a second draft pr attached to the ticket that I did which had a comment with some sql that splits apart the array ifnjson into rows.

Reply:
ah okay the json slq query is back in and even faster!